### PR TITLE
docs: finalise `docs-snippets`

### LINKED ISSUE #3205: Remove old `docs-snippets` directory
After all the snippets are ported to `docs-snippet2`, we should:
 - Remove `docs-snippets` directory
 - Move the snippets from `docs-snippets2` to the `docs` app to be alongside the markdown files
 - Remove `docs-snippets2` directory

### PR DESCRIPTION
**Pull request by petertonysmith94**
- Closes #3205

# Summary

- Finalized the migration of the `docs-snippets` to `docs`
- Related PR's
  - https://github.com/FuelLabs/docs-hub/pull/505
  - https://github.com/FuelLabs/github-actions/pull/48

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

**Comment by petertonysmith94** [comment]
We need to make some changes to the `docs-hub` GitHub workflow as it's currently failing in [CI](https://github.com/FuelLabs/fuels-ts/actions/runs/11816913325/job/32921229805#step:6:1930).

I'm opening this for review as it should be isolated to the [`github-actions` repository](https://github.com/fuellabs/github-actions/).

File: .github/workflows/code-lint.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ concurrency:
 
 jobs:
   docs:
-    uses: FuelLabs/github-actions/.github/workflows/vp-docs.yml@master
+    uses: FuelLabs/github-actions/.github/workflows/vp-docs.yml@ps/chore/relative-inline-snippets
```
Comment (petertonysmith94) [inline .github/workflows/code-lint.yaml:15]:
```suggestion
    uses: FuelLabs/github-actions/.github/workflows/vp-docs.yml@master
```

After the review + merge of https://github.com/FuelLabs/github-actions/pull/48

**Review by Torres-ssf — APPROVED** [review]

File: apps/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## Building
+
+The build process will build out our snippets and the documentation.
+
+```sh
+pnpm build
+```
+
+The snippets are built into testable scripts from the `src/snippets` folder and generates the Typegen types into the `src/snippets/typegend` folder. All test scripts end with a `.test.ts` suffix.
```
Comment (maschad) [inline apps/docs/README.md:20]:
```suggestion
The build process generates testable scripts from the snippets and outputs them alongside the snippets. The corresponding Typegen types are also generated and outputted in the `src/snippets/typegend` folder. All test scripts end with a `.test.ts` suffix.
```

File: apps/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+## Testing
+
+This will build the snippets and run the generated tests. To test a specific environment (`node` or `browser`), the snippet should be named as `{name}.{environment}.test.ts`. e.g. `deploy-contract.node.test.ts`
```
Comment (maschad) [inline apps/docs/README.md:24]: We had [discussed this before ](https://github.com/FuelLabs/fuels-ts/pull/3297/files/088d76810196f7dc38cd3d086ff1163c5444d124#r1804062940)on a previous PR but essentially @danielbate made the point about us standardizing how we want to specify test environments by continuing to use the _group_ comments as opposed to using file suffixes. I think we should perhaps continue in that vain.

File: apps/docs/scripts/pretest.sh [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Check if node is already running at port 4000, if not start it
-# TODO: This is a temporary solution to avoid conflicts with the test node in docs-snippets2
+# TODO: This is a temporary solution to avoid conflicts with the test node in docs-snippets
```
Comment (maschad) [inline apps/docs/scripts/pretest.sh:2]: This should no longer be necessary, perhaps we can revert this to 8e80ba16a

```bash
# Kill anything running on port 4000
lsof -t -i:4000 | xargs -r kill

# Runs a node at port 4000
pnpm fuels node > /dev/null 2>&1 &

# Builds projects
pnpm fuels build

# Deploys projects (needed for loader bytecode)
pnpm fuels deploy

# Kills the node
lsof -t -i:4000 | xargs -r kill

# Checks for type errors
pnpm tsc --noEmit
```

Although I don't think the type check is necessary anymore post-#3381

File: apps/docs/scripts/wrap-snippets.ts [hunk 5]
```diff
@@ -116,6 +121,14 @@ export const wrapSnippet = (filepath: string) => {
...
 const dir = 'src/**';
 const src = `${dir}/*.ts`;
-const ignore = [`src/typegend/**`, `src/env.ts`, `src/transactions/new-api.ts`, `${dir}/*.test.ts`];
+const ignore = [
+  `src/typegend/**`,
+  `src/env.ts`,
+  `src/vite.config.ts`,
+  `src/versions.data.ts`,
+  `src/jsonc.d.ts`,
+  `src/guide/transactions/new-api.ts`,
```
Comment (maschad) [inline apps/docs/scripts/wrap-snippets.ts:130]: Not introduced on this PR but does this file exist? Not seeing it

File: package.json [hunk 6]
```diff
@@ -46,8 +46,8 @@
     "lint:package-jsons": "tsx ./scripts/lint-package-jsons",
     "type:check": "turbo run type:check",
     "type:check-tests": "tsc -p tsconfig.test.json",
-    "prettier:check": "prettier --check packages --check apps/docs-snippets",
-    "prettier:format": "prettier --write packages --check apps/docs-snippets",
+    "prettier:check": "prettier --check packages --check apps/docs",
+    "prettier:format": "prettier --write packages --write apps/docs",
```
Comment (maschad) [inline package.json:50]: Is this `--write`  intentional? 

```suggestion
    "prettier:format": "prettier --write packages --check apps/docs",
```

File: apps/docs/tsconfig.json [hunk 7]
```diff
@@ -1,8 +1,10 @@
 {
   "extends": "../../tsconfig.base.json",
   "compilerOptions": {
-    "outDir": "./dist"
+    "module": "ES2022",
```
Comment (maschad) [inline apps/docs/tsconfig.json:4]: Just confirming but this is to enable top-level await?

**Review by maschad — COMMENTED** [review]
Awesome job on this @petertonysmith94 , so happy to finally put these doc changes to bed 🚀 😃 

I've left some comments and questions.

File: apps/docs/src/guide/contracts/call-parameters.md [hunk 8]
```diff
@@ -13,7 +13,7 @@ When interacting with contracts, you can configure specific parameters for contr
```
[the remark this answers is not in the document]
Comment (petertonysmith94) [reply apps/docs/src/guide/contracts/call-parameters.md:13]: Given this PR's size and complexity, I intend to keep the snippets consistent with those in `master`. Please feel free to make any changes to snippets in a subsequent PR.

File: package.json (hunk 6 above)
Comment (petertonysmith94) [reply package.json:50]: Yes, I'd expect to fix the changes, as it's the `prettier:format` command.

File: apps/docs/tsconfig.json (hunk 7 above)
Comment (petertonysmith94) [reply apps/docs/tsconfig.json:4]: Correct :)

File: apps/docs/.spellcheck.yml [hunk 9]
```diff
@@ -1,33 +1,32 @@
...
-      dictionary:
-        encoding: utf-8
-        wordlists:
-          - apps/docs/spell-check-custom-words.txt
-      pipeline:
-        - pyspelling.filters.context:
-            context_visible_first: true
-            escapes: \\[\\`~]
-            delimiters:
+  - name: SPCheck
```
[the remark this answers is not in the document]
Comment (petertonysmith94) [reply apps/docs/.spellcheck.yml]: 0d9fbf07d4117fc6a50f839442135a4b22123b05

File: apps/docs/README.md (hunk 2 above)
Comment (petertonysmith94) [reply apps/docs/README.md]: 6b0b25fac334db850d032d28cd5b713438f9cb3f

File: apps/docs/README.md (hunk 3 above)
Comment (petertonysmith94) [reply apps/docs/README.md]: 6b0b25fac334db850d032d28cd5b713438f9cb3f

File: apps/docs/scripts/wrap-snippets.ts (hunk 5 above)
Comment (petertonysmith94) [reply apps/docs/scripts/wrap-snippets.ts]: f5f024c1fbca47fdec05d41c6461bf2c0e7d1b70

File: apps/docs/.spellcheck.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (petertonysmith94) [reply apps/docs/.spellcheck.yml]: This is caused by the addition of prettier for the `app/docs` directory, as the [CI is failing](https://github.com/FuelLabs/fuels-ts/actions/runs/11969389455/job/33370027628#step:9:27). I am going to revert this  and go ahead with the `yml` formatting :)

**Review by Torres-ssf — DISMISSED** [review]

File: .prettierignore [hunk 10]
```diff
@@ -14,8 +14,11 @@ apps/demo-typegen/src/script-types
 apps/demo-typegen/src/predicate-types
 apps/docs/.vitepress/cache/
 apps/create-fuels-counter-guide
-apps/docs-snippets/.fuels
-apps/docs-snippets2/.fuels
+apps/docs/.fuels
+apps/docs/.typedoc
+apps/docs/src/api
+apps/docs/src/**/*.ts
```
Comment (nedsalk) [inline .prettierignore:20]: This path is being ignored here but [added](https://github.com/FuelLabs/fuels-ts/pull/3369/files#diff-663ade211b3a1552162de21c4031fcd16be99407aae5ceecbb491a2efc43d5d2R29) in `.prettierrc`. I think one of them should go :smile:

Comment (petertonysmith94) [reply .prettierignore:20]: Great catch!

🦅  👀

**Review by nedsalk — APPROVED** [review]

File: .prettierignore (hunk 10 above)
Comment (petertonysmith94) [reply .prettierignore]: [9fbb033](https://github.com/FuelLabs/fuels-ts/pull/3369/commits/9fbb033a5ba0f9f788da5dcf5ffa165bc69bce07)

Comment (maschad) [reply .prettierignore]: I'd actually suggest ignoring at least the `test` files that get generated at least (potentially even the snippets as well)

**Review by nedsalk — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Pull request merged**